### PR TITLE
Fixed bugs in conditional checking

### LINKED ISSUE #1728: Inappropriate conditional checking
- Operating System:
- Node Version:8.14.0
- NPM Version:6.7.0
- webpack Version:4.28.4
- webpack-dev-server Version:3.1.14

- [x] This is a **bug**
- [ ] This is a **modification** request

### Expected Behavior
When requested for the script by the user which is not present empty script should be returned in `getCurrentScriptSourceSimilar` function. Similar problem with `overlay` function and `error` function.

### Actual Behavior
When user requests for a script which is not present then nothing happens.

### For Bugs; How can we reproduce the behavior?
In the server try getting the script or do something which produces error.

### PR DESCRIPTION
**Pull request by abhinavsagar**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
No

[Fixes #1728 ](https://github.com/webpack/webpack-dev-server/issues/1728)

### Breaking Changes
No

### Additional Info
No

### CODE REVIEW

**Comment by abhinavsagar** [comment]
@hiroppy @evilebottnawi I am not quite sure how to write tests for these. Can you help me in writing tests for my changes?

**Comment by evilebottnawi** [comment]
@abhinavsagar no need, right now, anyway can you create minimum reproducible test repo for manual testing?

**Comment by abhinavsagar** [comment]
@hiroppy @evilebottnawi I tried writing minimum reproducible test but was not able to with my limited knowledge. Can you help me. Is it not possible to check correctness without writing the tests.

**Comment by evilebottnawi** [comment]
@abhinavsagar no, we can't merge this without tests or minimum reproducible test repo

File: client-src/default/index.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ function getCurrentScriptSource() {
     return document.currentScript.getAttribute('src');
   }
   // Fall back to getting all scripts in the document.
-  const scriptElements = document.scripts || [];
+  const scriptElements = document.scripts ? document.scripts : [];
+
```
Comment (cyjake) [inline client-src/default/index.js]: what's the point of this change?

**Comment by evilebottnawi** [comment]
Closing due to inactivity. Thanks! If you still this problem please open new issue with reproducible test repo

**Pull request closed without merging**